src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -104,30 +81,54 @@ class JesBackend extends Backend with LazyLogging {
     }
   }
 
+  sealed trait JesParameter {
+    def name: String
+    def gcs: String
+    def local: Path
+    def parameterType: String
+
+    final val isInput = this.isInstanceOf[JesInput]
```

Comment:
knowledge of subtypes is a common pattern in scala adts fwiw. for instance implementing polymorphism via a pattern match in trait using all subtypes